tools/renumber-migrations

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import glob
+import os
+import sys
+import fileinput
+import re
+
+def validate_order(order, length):
+    if len(order) != length:
+        print("Please enter the sequence of all the conflicting files at once")
+        sys.exit(1)
+
+    for i in order:
+        if i > length or i < 1 or order.count(i) > 1:
+            print("Incorrect input")
+            sys.exit(1)
+
+def renumber_migration(conflicts, order, last_correct_migration):
+    stack = []
+    for i in order:
+        if conflicts[i-1][0:4] not in stack:
+            stack.append(conflicts[i-1][0:4])
+        else:
+            # Replace dependencies with the last correct migration
+            file = fileinput.FileInput('zerver/migrations/' + conflicts[i-1], inplace=True)
+            for line in file:
+                print(re.sub(r'[\d]+(_[a-z0-9]+)+', last_correct_migration, line), end='')
+
+            # Rename the migration indexing at the end
+            new_name = conflicts[i-1].replace(conflicts[i-1][0:4], '00' + str(int(last_correct_migration[0:4]) + 1))
+            os.rename('zerver/migrations/' + conflicts[i-1], 'zerver/migrations/' + new_name)
+
+            last_correct_migration = new_name
+
+def resolve_conflicts(conflicts):
+    print("Conflicting migrations:")
+    for i in range(0, len(conflicts)):
+        print(str(i+1) + '. ' + conflicts[i])
+
+    order = raw_input("Enter the order in which these migrations should be arranged: ")
+    order = map(int, order.split())
+    validate_order(order, len(conflicts))
+    last_correct_migration = str(raw_input('Enter the name of last correct migration file: '))
+    renumber_migration(conflicts, order, last_correct_migration)
+
+if __name__ == '__main__':    
+
+    while True:
+        conflicts = []
+        stack = []
+        files_list = [os.path.basename(path) for path in glob.glob("zerver/migrations/????_*.py")]
+        file_index = [file[0:4] for file in files_list]
```

Comment:
Is the index always made up of the first four characters? If it can grow larger, it might be worth making the 4 a variable (if I understood this piece of code correctly).